proposal.md

```diff
@@ -0,0 +1,5 @@
+# Project Gutenberg In a Box Proposal
+
+We are Dylan DiGeronimo (ddigeron@stevens.edu), Ryan Locke (rlocke@stevens.edu), Naseem Miah (nmiah@stevens.edu), and Taleen Mitchell (tmitche2@stevens.edu), and we are extremely excited to have the chance to work with the Free Ebook Foundation to bring Project Gutenberg to more and more people around the world. The five of us posess a broad skillset, including experience with Linux systems, front and back end web development, database configuration, interface design, technical writing and documentation, and cybersecurity.
+
+Our end goal in this project is to produce a workable demonstration of the Project Gutenberg in a Box module, running on a RaspberryPi minicomputer. This system will be able to be easily deployed as an Internet in a Box module, and upon deployment will use Ansible to automatically install and setup various components, which will come together as a whole to form the PG in a Box system. Among these components will be a web server, hosting an easy to use web application that will allow users to interact with the library system. Another will be a database system containing the full PG library, which will allow users to search the system as well as find books based on categories, including genre. Additionally, as many of our module's users will not have regular access to broadband internet, we will create a full suite of documentation to simply illustrate how to make full use of the system. Finally, as our team contains students passionate about cybersecurity, we also plan to make the security of our system a top priority, and will ensure the privacy and integrity of any user information, as well as making sure to mitigate any vulnerabilities introduced by our module.
```

Comment:
What are the security affordances of Internet-in-a-box? Typical deployment environments won't have access to the usual certificate infrastructure. (I don't know the answer.)

Reply:
We assumed that a situation could arise where the module could be tampered with to pass a payload to users who connect to it. This could be especially problematic if the module was to be deployed in environments where the RaspberryPi might have an internet connection, for example in a university, school, or public library

Reply:
http://wiki.laptop.org/go/IIAB/FAQ#What_security_tips_exist.3F

Reply:
http://wiki.laptop.org/go/IIAB/FAQ#Captive_Portal_Administration:_What_tips_.26_tricks_exist.3F

Reply:
The security issues surrounding disconnected servers, especially when they interact with user agents that rely on being connected to the internet (see the android issue with captiveportals in the link above) are mostly neglected by the internet security community, and thus might be an excellent topic for a student. It's a bit different from the main thrust of this project, but a proposal that sensible incorporates that sort of activity would be stronger for it.